### ISSUE TITLE
Cannot create load balancer in AWS

### ISSUE THREAD

**Issue by texasmichelle** [kubicorn/kubicorn#656]
I'm seeing the following error when installing Kubeflow on a cluster generated with Kubicorn:

```
  Normal   EnsuringLoadBalancer        2m (x19 over 1h)   service-controller  Ensuring load balancer
  Warning  CreatingLoadBalancerFailed  2m (x10 over 47m)  service-controller  (combined from similar events): Error creating load balancer (will retry): failed to ensure load balancer for service kubeflow-chasm/tf-hub-lb: AccessDenied: User: arn:aws:sts::015345740490:assumed-role/kubeflow-demo-KubicornMasterRole/i-0c953d8dc191ce89e is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::015345740490:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
           status code: 403, request id: b6003393-8fb1-11e8-aebb-c1258eb69b8d
```

The user I created to run kubicorn has IAMFullAccess.

**Comment by stevenoctopus** [comment]
I'm not using kubicorn but I just ran into this today walking through the Getting Started with Amazon EKS guide. I followed the same exact steps I did last week, where I did not receive this error.

I was able to get around it by attaching a new policy to the role I assigned to my EKS cluster but I wonder what changed?

**Comment by texasmichelle** [comment]
@stevenoctopus Which policy fixed the issue?

**Comment by stevenoctopus** [comment]
I created a new policy, which allowed the service role I created for eks to perform iam:CreateServiceLinkedRole on the AWSServiceRoleForElasticLoadBalancing role in my account, and then attached that to the eks service role.

However, I don't think that is correct to do. I found out AWS automatically created the AWSServiceRoleForElasticLoadBalancing role if you created a load balancer before January of this year. (I'm on my phone or I would link the docs where I found this)

Nothing in my account was using the AWSServiceRoleForElasticLoadBalancing role so I wanted to try deleting it and starting the process over. I will try on Monday and report back my results.

**Comment by stevenoctopus** [comment]
No luck deleting the role and trying again. I get the same error even when the AWSServiceRoleForElasticLoadBalancing role doesn't exist in my account. Very strange. 

There is a [post](https://forums.aws.amazon.com/thread.jspa?threadID=286810&tstart=0) about this in the AWS forums that I am also watching.

**Comment by stevenoctopus** [comment]
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "iam:CreateServiceLinkedRole",
            "Resource": "arn:aws:iam::015345740490:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing"
        }
    ]
}
```

Here is a policy you can attach to the EKS service role as a workaround for the time being. I subbed in the correct account ID for you already.

**Comment by texasmichelle** [comment]
Thanks for posting the additional context. Is `iam:CreateServiceLinkedRole` not covered by `IAMFullAccess`?

**Comment by stevenoctopus** [comment]
It is, but it would be unwise to give EKS IAMFullAccess as it could
possibly delete/modify existing IAM services.

**Comment by texasmichelle** [comment]
Makes sense - this particular example is for a demo, so I'm less concerned for now & can reduce the scope once it's all working. The service account I created for kubicorn already has IAMFullAccess, so I'm not sure I understand how to apply this workaround.
